hw3/hw3-react/src/CharactersRequest.js

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react';
+import axios from 'axios';
+const url = 'https://thronesapi.com/api/v2/Characters';
+
+function CharactersRequest() {
+  const getCharacters = async function getCharactersFromAPI() {
+    const response = await axios(url);
+    return response;
+  };
+
+  useEffect(() => {
+    getCharacters()
+      .then((data) => {
+        sessionStorage.setItem('characters', JSON.stringify(data));
```

Comment:
Separating the logic for the API request to retrieve the data make sense for modularity which is what I did as well. However, I learned from the TA that making the request in the 'parent' component  (which your case would be the AppRouter.js file) is more efficient because you can then pass the data as a prop when you call the other components like Search and HouseChart instead of using sessionStorage.

Reply:
Yeah I learned a little more about the concepts about state management and best practices in react a bit too late lol. I'll get to fixing this following the logic of your feedback. Thanks!